### PR TITLE
#257 warn of unspecified field arguments

### LINKED ISSUE #257: Inspection: warn if user uses rule/checkpoint object in input instead of rules section
E.g. quite hard to find an error in large project, which is actually in missing ".input" in latest rule dependency

image

Snakemake returns smth meaningless:
```
Full Traceback (most recent call last):
  File "/home/user/miniconda3/envs/snakemake/lib/python3.7/site-packages/snakemake/rules.py", line 463, in _set_inoutput_item
    for i in item:
TypeError: 'RuleProxy' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/user/miniconda3/envs/snakemake/lib/python3.7/site-packages/snakemake/__init__.py", line 404, in snakemake
    print_compilation=print_compilation)
  File "/home/user/miniconda3/envs/snakemake/lib/python3.7/site-packages/snakemake/workflow.py", line 743, in include
    exec(compile(code, snakefile, "exec"), self.globals)
  File "/mnt/stripe/rcherniatchik/_playground/pipeline/sources2/Snakefile", line 91, in <module>
  File "/home/user/miniconda3/envs/snakemake/lib/python3.7/site-packages/snakemake/workflow.py", line 815, in decorate
    rule.set_input(*ruleinfo.input[0], **ruleinfo.input[1])
  File "/home/user/miniconda3/envs/snakemake/lib/python3.7/site-packages/snakemake/rules.py", line 254, in set_input
    self._set_inoutput_item(item)
  File "/home/user/miniconda3/envs/snakemake/lib/python3.7/site-packages/snakemake/rules.py", line 470, in _set_inoutput_item
    "Input and output files have to be specified as strings or lists of strings.")
  File "<string>", line None
SyntaxError: Input and output files have to be specified as strings or lists of strings.

SyntaxError:
Input and output files have to be specified as strings or lists of strings.
  File "/mnt/stripe/rcherniatchik/_playground/pipeline/sources2/Snakefile", line 78, in <module>
```

### PR DESCRIPTION
**Pull request by SimonTsirikov**
Fixes #257

### CODE REVIEW

File: src/main/resources/SnakemakeBundle.properties [hunk 1]
```diff
@@ -38,6 +38,10 @@ INSP.NAME.section.multiline.string.args.message=Multiline string argument in ''{
 INSP.NAME.section.multiple.args=Only one argument is allowed for this section.
 INSP.NAME.section.multiple.args.message=Only one argument is allowed for ''{0}'' section.
 
+# SmkSectionMistypedArgsInspection
+INSP.NAME.section.unspecified.field.args=Field is not specified
+INSP.NAME.section.unspecified.field.args.message=No field from ''{0}'' is specified
```
Comment (iromeo) [inline src/main/resources/SnakemakeBundle.properties:43]: Could you change the message to smth more clear, so user could easily understand what `field` you are talking about and how to fix the bug. E.g. smth like: Expected rule section name or argument (e.g. 'rules.foo.input' or 'rules.foo.output.arg1'), but  rule/checkpoint/subworkflow object was found

File: src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkSectionUnspecifiedFieldArgsInspection.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+        override fun visitPyBinaryExpression(node: PyBinaryExpression) {
+            node.acceptChildren(this)
+        }
+
+        override fun visitPyParenthesizedExpression(node: PyParenthesizedExpression) {
+            node.acceptChildren(this)
+        }
+
+        override fun visitPyReferenceExpression(node: PyReferenceExpression?) {
```
[the remark this answers is not in the document]
Comment (SimonTsirikov) [reply src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkSectionUnspecifiedFieldArgsInspection.kt]: Fixed.

File: src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkSectionUnspecifiedFieldArgsInspection.kt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            if (node?.firstChild?.textMatches("rules") == true) {
```
[the remark this answers is not in the document]
Comment (SimonTsirikov) [reply src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkSectionUnspecifiedFieldArgsInspection.kt]: Done.

File: src/main/resources/SnakemakeBundle.properties (hunk 1 above)
Comment (SimonTsirikov) [reply src/main/resources/SnakemakeBundle.properties]: Done.

**Pull request merged**